src/_pytest/python.py

```diff
@@ -267,22 +267,18 @@ def pytest_pycollect_makeitem(collector, name, obj):
             outcome.force_result(res)
 
 
-class PyobjMixin:
+class PyobjMixin(nodes.Node):
```

Comment:
I think this is not a good idea in general -- it makes it look like `PyobjMixin` is an actual `Node`.

TBH I'd prefer to get rid of `PyobjMixin` entirely (and this looks like a positive step towards that), but in my typing branch I solved it like this for now:

https://github.com/bluetech/pytest/commit/a34e7ff53e7d63c38ba0b4663c7564b68281f5b4#diff-88cb3736bed2f144d42f07ccbeb97eb0R280-R291

Reply:
> I think this is not a good idea in general -- it makes it look like `PyobjMixin` is an actual `Node`.

Agreed!

> TBH I'd prefer to get rid of `PyobjMixin` entirely (and this looks like a positive step towards that), but in my typing branch I solved it like this for now:
> 
> [bluetech@a34e7ff#diff-88cb3736bed2f144d42f07ccbeb97eb0R280-R291](https://github.com/bluetech/pytest/commit/a34e7ff53e7d63c38ba0b4663c7564b68281f5b4#diff-88cb3736bed2f144d42f07ccbeb97eb0R280-R291)

Thanks - took that approach here also.

Reply:
Note however that it should be `parent: PyobjMixin` with `PyobjMixin` really (since it assumes that `parent` has `obj`), which does not work without subclassing `Node` (`Definition of "parent" in base class "PyobjMixin" is incompatible with definition in base class "Node"`).  And using `parent: Optional[Union["Node", "PyobjMixin"]] = None,` with `Node` results in `Item "PyobjMixin" of "Union[Node, PyobjMixin]" has no attribute "config"` etc.
So I think that subclassing `Node` here is the best approach for now.
Despite its name it is a base/intermediate class for all "python" nodes/collectors after all.

Reply:
> Note however that it should be parent: PyobjMixin with PyobjMixin really

I guess such constraints on `parent` can be added once the `from_parent` work is completed? For now however I'm not sure they're valid. It's definitely an issue that typing expose.

> Despite its name it is a base/intermediate class for all "python" nodes/collectors after all.

Maybe changing it to a common base for all "python" nodes makes sense, I am not sure. One problem with that is that it breaks the clear `Item/Collector` split, since both types will derive from it. And might confuse MRO/`from_parent` a lot too...

Anyway, as long as it is a mixin and written as a mixin, IMO it is better not to derive from `Node`. And we should strive to eliminate it as such :)